### PR TITLE
Update developer documentation

### PR DESCRIPTION
**Pull request by karlaspuldaro**
This PR updates the developer-workflow documentation.

- Adds dev requirements
- Adds installation section
- Adds section for setting up elyra fork and upstream
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: docs/source/developer_guide/development-workflow.md [hunk 1]
```diff
@@ -18,15 +18,58 @@ limitations under the License.
...
+* [Yarn](https://yarnpkg.com/lang/en/docs/install)
+
+### Setting up your development environment
+
+* Install Miniconda
+Download and install a [Python 3 version of Miniconda](https://docs.conda.io/en/latest/miniconda.html) according to your Operating System
+
+* Create a new Python environment
+
+    `conda create -n <env-name> python`
```
Comment (ptitzler) [inline docs/source/developer_guide/development-workflow.md:34]: Could you switch to block quotes to keep formatting consistent?

```
conda create -n <env-name> python
```

File: docs/source/developer_guide/development-workflow.md [hunk 2]
```diff
@@ -18,15 +18,58 @@ limitations under the License.
...
+
+    The python version of your environment will match the miniconda version you installed. You can override the default by explicitly setting `python=3.7`, for example.
+
+* Activate the new environment
+
+    `conda activate dev`
```
Comment (ptitzler) [inline docs/source/developer_guide/development-workflow.md:40]: See previous comment + please  use the placeholder (throughout the doc) or an example name (e.g. `dev`)

```
    conda activate <env-name>
```

File: docs/source/developer_guide/development-workflow.md [hunk 3]
```diff
@@ -18,15 +18,58 @@ limitations under the License.
...
+    which python # Displays current python path
+    pip --version
+    which pip
+    ```
+    Python path must be under miniconda envs folder.
+    Confirm pip location matches where miniconda is installed.
+
+* Install NodeJS
+
+    `conda install -y -c conda-forge/label/main nodejs`
```
Comment (ptitzler) [inline docs/source/developer_guide/development-workflow.md:55]: Block quote

```
conda install -y -c conda-forge/label/main nodejs
```

**Review by ptitzler — COMMENTED** [review]
Looking good aside from minor formatting inconsistencies

File: docs/source/developer_guide/development-workflow.md (hunk 2 above)
Comment (karlaspuldaro) [reply docs/source/developer_guide/development-workflow.md]: Good catch! I forgot to update that one with the same placeholder as above

**Review by lresende — APPROVED** [review]

**Pull request merged**